comms/core/src/peer_manager/node_identity.rs

```diff
@@ -49,6 +49,8 @@ pub struct NodeIdentity {
     public_key: CommsPublicKey,
     features: PeerFeatures,
     secret_key: CommsSecretKey,
+    stealth_address_scanning_private_key: CommsSecretKey,
```

Comment:
`NodeIdentity` is used for all network participants (base nodes, validator nodes etc) and the `comms` crate is for general purpose p2p networking so putting stealth address keys in the NodeIdentity is not appropriate.

Reply:
Yes. I'll close this PR, since this actually super complicates things. I will have to think this over.